src/act.other.cpp

```diff
@@ -4165,6 +4165,128 @@ ACMD(do_memory)
   }
 }
 
+// Looks for any working cyberdeck in the character's inventory or worn equipment.
+struct obj_data *find_cyberdeck(struct char_data *ch)
+{
+  struct obj_data *deck;
+  if (!ch) {
+    mudlog("SYSERR: Got NULL ch to find_cyberdeck()!", ch, LOG_SYSLOG, TRUE);
+    return NULL;
+  }
+
+  for (deck = ch->carrying; deck; deck = deck->next_content) {
+    if (OBJ_IS_VALID_CYBERDECK(deck)) {
+      return deck;
+    }
+  }
+
+  for (int wear_idx = 0; !deck && wear_idx < NUM_WEARS; wear_idx++) {
+    if (OBJ_IS_VALID_CYBERDECK(GET_EQ(ch, wear_idx))) {
+      return GET_EQ(ch, wear_idx);
+    }
+  }
+
+  return NULL;
+}
+
+ACMD(do_download_headware)
+{
+  struct obj_data *deck, *jack, *memory;
+  skip_spaces(&argument);
+  if (!*argument) {
+    send_to_char("Which file number do you want to transfer out of headware memory?\r\n", ch);
+    return;
+  }
+
+  for (memory = ch->cyberware; memory; memory = memory->next_content) {
+    if (GET_OBJ_VAL(memory, 0) == CYB_MEMORY) {
```

Comment:
Nit: This is correct, but for readability please use `GET_CYBERWARE_TYPE()` instead of a direct objval reference. I'm slowly replacing those old references as I see them in the code, but haven't gotten to them all yet.